### PR TITLE
chore: .gitignore should be a whitelist

### PR DESCRIPTION
**Pull request by thengoster**
Pr for issue #472

Replaced .gitignore with deny-by-default version by first ignoring everything, and then whitelisting the directories/files we want in our git repository.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,21 +1,54 @@
...
+!/SECURITY.md
+!/SUMMARY.md
+!/changelog.md
+!/code-of-conduct.md
+!/jest.config.js
+!/lerna.json
+!/package.json
+!/tsconfig.json
+!/vercel.json
+!/yarn.lock
```
Comment (mrnerdhair) [inline .gitignore:54]:
```suggestion
!/yarn.lock

```

File: .gitignore [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
+!/integration/src/**
+
+!/packages/
+!/packages/*
+!/packages/**/src/
+!/packages/**/src/**
+!/packages/**/.npmignore
+!/packages/**/jest.config.js
+!/packages/**/package.json
+!/packages/**/tsconfig.json
```
Comment (mrnerdhair) [inline .gitignore:29]:
```suggestion
!/packages/*/src/
!/packages/*/src/**
!/packages/*/.npmignore
!/packages/*/jest.config.js
!/packages/*/package.json
!/packages/*/tsconfig.json
```
(haven't tested this yet, but we probably don't want to apply these recursively)

**Review by mrnerdhair — APPROVED** [review]
Personally I think this is better sorted, but I'll leave it up to reviewer number 2; I also made .prettierignore a symlink so we don't have to worry about it getting out of sync anymore.

**Comment by pastaghost** [comment]
LGTM. I just suggest that we update the README with some text for future developers indicating that newly added root-level files/subdirectories won't be tracked by default and exceptions will need to be added to the .gitignore.

**Review by pastaghost — APPROVED** [review]

**Pull request merged**